compiler-core/src/exhaustiveness.rs

```diff
@@ -290,6 +298,33 @@ impl<'a> Compiler<'a> {
     }
 
     fn compile_rows(&mut self, rows: Vec<Row>) -> Decision {
+        // If there are no rows, we get an immediate decision failure,
```

Comment:
There's still a `if rows.is_empty() {` in do_compile. How about we move the true branch from this if/else into a new method (handle_empty_row or something?) and then merge `compile_rows` and `do_compile_rows` back together.

Reply:
The entire algorithm sort of relies on that `Failure` being returned in `do_compile_rows`, it terminates the recursive process. If we try to continue compiling at that point, we run into an infinite loop of failing and trying to add more info, but failing again. This only really works if we do it once at the beginning. I'm not really sure if there's a way to rework this without looping forever

Reply:
Ah I see, thank you. Given these are non-trivial functions can we give them better names?